Add support in the encoder for up to 20 bit sample rates

ckolivas opened the pull request:
Add support in the encoder for up to full 20 bit sample rates now that many DACs support up to 768kHz. Files produced with these sample rates are already decodable by the reference decoder. Update documentation and tests accordingly.

ValZapod commented:
I suppose this will need testing?

ktmf01 commented on src/test_libFLAC/format.c:
```diff
@@ -65,11 +65,14 @@ static struct {
+	{ 705600 , true , true  },
+	{ 768000 , true , true  },
+	{ 1000000, true , true  },
+	{ 1048575, true , false },
```
I think all these true's in the last column should be false. The name of this column is 'subset', and these new sample rates aren't subset. If that makes the test fail, there might be some other code that needs fixing

ktmf01 commented:
This needs to be changed too:
https://github.com/xiph/flac/blob/37d1a620eb2c2ffe52446df5832608196d57f562/src/libFLAC/format.c#L219-L232

ktmf01 commented:
@ckolivas I hope you don't mind that I squashed and rebased this PR and reflowed the commit message. The checks have changed since you sent in this PR, and I wanted the tests rerun before this is merged.

ckolivas commented:
> @ckolivas I hope you don't mind that I squashed and rebased this PR and reflowed the commit message. The checks have changed since you sent in this PR, and I wanted them rerun before this is merged.

That's fine, I haven't been attending to code in a long time so appreciate you taking it up.

ValZapod commented:
This needs to be updated now. https://xiph.org/flac/faq.html#general__samples

>FLAC supports linear sample rates from 1Hz - 655350Hz in 1Hz increments.

To 
>FLAC supports linear sample rates from 1Hz - 1048575Hz in 1Hz increments.

ktmf01 commented:
I think it is better to wait until this is included in a release
